src/packages/dca/api/dcaApi.ts

```diff
@@ -313,36 +314,37 @@ export class DcaApi {
 		type: "active" | "past";
 	}): Promise<DcaOrderObject[]> => {
 		const { type, walletAddress } = inputs;
-		try {
-			const uncastedResponse =
-				await this.Provider.indexerCaller.fetchIndexer<
-					DcaIndexerOrdersResponse,
-					DcaIndexerOrdersRequest
-				>(
-					`dca/get/${type}`,
-					{
-						sender: walletAddress,
-					},
-					undefined,
-					undefined,
-					undefined,
-					true
+		const uncastedResponse = await this.Provider.indexerCaller.fetchIndexer<
+			DcaIndexerOrdersResponse,
+			DcaIndexerOrdersRequest
+		>(
+			`dca/get/${type}`,
+			{
+				sender: walletAddress,
+			},
+			undefined,
+			undefined,
+			undefined,
+			true
+		);
+		const orders = uncastedResponse.orders
+			.sort((lhs, rhs) => rhs.created.timestamp - lhs.created.timestamp)
+			.map((order) => Casting.dca.createdOrderEventOnIndexer(order));
+
+		await Promise.all(
+			orders.map(async (order) => {
+				await Promise.all(
+					order.trades.map(async (trade) => {
+						trade.rate = await this.getRate(trade);
```

Comment:
this is not the right way to do this -- you should never update a variable in map outside of its scope . instead save result of promise.alls into a variable and then assign these values